wandb/sdk/internal/artifacts.py

```diff
@@ -146,6 +146,7 @@ def save(
             # TODO: update aliases, labels, description etc?
             if use_after_commit:
                 self._api.use_artifact(artifact_id)
+            self._cleanup_staged_entries()
```

Comment:
should this be in a `try/finaly`?

Reply:
The actual cleanup (the `os.remove` calls) are in a `try` block; what's the `finally` you would like to see?

Reply:
I meant the entire `save()` method, i.e. we can do something like renaming this to `_save_internal()` and write:
```
def save():
  try:
     self._save_internal()
  finally:
     self._cleanup..
```
main benefit is that if someone adds new control flow in `save()` we wont worry about missing the clean up

Reply:
Ah, got it. Sure, that makes sense.

Since I didn't want to repeat the humongous call signature, I added this as a decorator to `save`. If you feel like that's an abuse of a decorator (or just an unnecessary use of one) I'm happy to change it to be a more straightforward nesting.

Reply:
I take it back. `mypy` really didn't like a class method based decorator and I didn't want to mess with the annotation on the wrap function either, so I scrapped that and am just calling through to `_save_internal` as per your initial suggestion.